subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationState.java

```diff
@@ -17,7 +17,7 @@
 package org.gradle.internal.progress;
 
 public interface BuildOperationState {
-    Object getId();
+    OperationIdentifier getId();
```

Comment:
@jjohannes I would like your awareness of this change. 

I moved `OperationIdentifier` into `base-services` package `org.gradle.internal.progress`. This fixes one of the problematic dependencies with ProgressLogging. 

I decided it best to introduce a strong type here now that the class is available so that I could manage this as an `OperationIdentifier` instead of a plain `Object` in `DefaultBuildOperationExecutor`. 

Please acknowledge, and optionally, provide feedback on the parts of this PR that touch Build Operations and Workers infrastructure.